woocommerce/woocommerce-admin
# Show/hide (extension) task list and move completed tasks to bottom

Fixes #6638.

This PR sorts the extension task list by completion status and allows for toggling the list visibility via the display options menu.

Additionally, the list title has been changed to "Things to do next" and the "view" tracking event name has been corrected (there was a duplicate event name for each list).

In a potentially controversial move 🌶️  I added a SlotFill to the Display Options menu in order to implement the conditional visibility of the "show things to do next" item. (See: https://github.com/woocommerce/woocommerce-admin/issues/6638#issuecomment-819574780) I think SlotFill is a cleaner approach than connecting the `<DisplayOptions>` component to everything that determines what tasks are shown.

### Accessibility




### Screenshots

![Screen Shot 2021-04-13 at 9 52 30 AM](https://user-images.githubusercontent.com/63922/114573503-6c16a800-9c46-11eb-960f-ee3682383b65.png)
_Completed tasks at the bottom_

![Screen Shot 2021-04-13 at 9 52 51 AM](https://user-images.githubusercontent.com/63922/114573574-7c2e8780-9c46-11eb-8dfb-8e48e6d565b0.png)
_Visibility toggle_


### Detailed test instructions:

- Ensure you have multiple extension tasks (use this diff if you'd like: https://gist.github.com/jeffstieler/32cf7e591173e9d17c47f0a6ee2ac7e6)
- Select the "Example 1" task and mark it complete
- Verify that it is sorted to the bottom of the task list
- Click "display options"
- Toggle the "things to do next" list off
- Verify the list is no longer visible
- Verify the `wcadmin_extended_tasklist_hide` event is fired
